.github/workflows/unit_mac.yml

```diff
@@ -47,7 +47,15 @@ jobs:
       - run: bash scripts/environment/bootstrap-macos.sh
       - run: bash scripts/environment/prepare.sh
       - run: echo "::add-matcher::.github/matchers/rust.json"
-      - run: make test
+
+      # Some tests e.g. `reader_exits_cleanly_when_writer_done_and_in_flight_acks` are flaky.
+      - name: Run tests
+        uses: nick-fields/retry@v3
+        with:
+          timeout_minutes: 30
```

Comment:
may need to bump this up?  Looking at some of failure history, it looks like `unit-mac` takes ~1hr to run.